fashion-messaging-app-api/server.js

```diff
@@ -139,6 +86,71 @@ app.get('/posts', async (req, res) => {
     res.status(500).json({ message: err.message });
   }
 });
+app.get('/photos', async (req, res) => {
+  const query = req.query.search || defaultQuery;
+
+  
+  try {
+    const response = await clientAPI.photos.search({ query, per_page: PHOTOS });
+    
+    let fetchPromises = response.photos.map(async photo => {
+      if (cache.has(photo.src.original)) {
+        return cache.get(photo.src.original);
+      
+      } else {
+        var myHeaders = new Headers();
+        myHeaders.append("x-api-key", "7008b5da328806c39a5561c3d51339c18dbb66dcd8bbfeebca80f0a96c44c332");           
+                     
+
+        var formdata = new FormData();
+        formdata.append("image_url", photo.src.original);
+      
+        var requestOptions = {
+          method: 'POST',
+          headers: myHeaders,
+          body: formdata,
+        };
+
+        try {
+          const fetchResponse = await fetch('https://cloudapi.lykdat.com/v1/detection/items', requestOptions);
+         
+          if (fetchResponse.status === 400) {
+            throw new Error('Image size too large for the API.');
+          }
+          const result = await fetchResponse.json();
+
+          if (result.data.detected_items.length > 0) {
+            cache.set(photo.src.original, photo);
+            return photo;
+          }
+
+        } catch (error) {
+          console.error('error', error);
+          return null;
+        }
+      }
+    });
+
+    const fashionPhotos = await Promise.all(fetchPromises);
+    const validPhotos = fashionPhotos.filter(photo => photo !== undefined && photo !== null);
+
+    for (let photo of validPhotos) {
+      await SearchResult.create({
+        searchTerm: query,
+        imageUrl: photo.src.original,
+        isFashion: true,  
+        userId: req.session.user ? req.session.user.id : null,  
+      });
+    }
```

Comment:
This helps me add the user searchResults to the searchResults table as the rows and columns in my database

Reply:
is isFashion being set differently anywhere else? 
do you have a plan to set isFashion according to the search query?
if this is always going to be set to true, do we need to include the param or could it be removed?

Reply:
> is isFashion being set differently anywhere else?
> do you have a plan to set isFashion according to the search query?
> if this is always going to be set to true, do we need to include the param or could it be removed?

I plan to to use it to store isFashion as a boolean in my database. I will better define it later instead of the hardcoded "true"